### ISSUE TITLE
A lot of unwanted fonts in selection

### ISSUE THREAD

**Issue by khalid-hussain** [vanniktech/VNTFontListPreference#5]
Hi there,

I recently had a problem with this library as it was showing me a lot of fonts which I did not put into the fonts directory. I used this library in my app which can be found here and the issue highlighted here.

So I came here and saw that you had updated your library and so I updated it too hoping to have this issue fixed. Before attempting it in my app, I decided to try the demo app to verify if it was the library or my implementation.

My Device:
Samsung Galaxy Note II N7100 running Cyanogenmod Lollipop 5.1.1

Here are the screenshots:
![font_1](https://cloud.githubusercontent.com/assets/3102714/8762623/fb3faf0a-2db0-11e5-935c-5e7f920a8994.png)
![font_2](https://cloud.githubusercontent.com/assets/3102714/8762622/fb3f21ac-2db0-11e5-927b-063347789dd4.png)

After a little debugging, I found that the problem lies in naming the font assets directory as 'fonts'. If the name is different such as in the demo app, i.e. 'fonts_2', the problem does not exist. This is not a problem but rather an inconvenience and I think it should be mentioned in the README.

Thanks.

**Comment by vanniktech** [comment]
I'm not quite sure what you mean. Though I've improved the explanation of the fontDirectory attribute in the README file. Is that good enough for you or is there anything else?

**Comment by khalid-hussain** [comment]
If you look at the fonts directory over [here](https://github.com/vanniktech/VNTFontListPreference/tree/master/app/src/main/assets/fonts), you will see that there are only 3 fonts. But when you run the demo app, it shows more than 3 fonts as shown in the screenshot. I hope it is clear now?

**Comment by vanniktech** [comment]
Alright I'll take a look at it later.

**Comment by vanniktech** [comment]
Okay, so I know what you mean, though I was not able to reproduce it. However I refactored the class and wrote some tests around it. 898bab36b97294e9502f93f52aadd60b9ad71844

Could you please try it out whether the problem still exists or not?

**Comment by khalid-hussain** [comment]
It still exists.

**Comment by vanniktech** [comment]
Could it be related to your device / Android 5.1.1? I've tested the app itself on various devices and I could not encounter that issue.

**Comment by vanniktech** [comment]
@khalid-hussain 
I've released version 0.1.2 of the library which fixes some things. Could you retry it with that version in your project?

**Comment by khalid-hussain** [comment]
I'm sorry but my phone just died the other day. It restarted and got stuck in a boot loop. I've tried re-flashing even through ODIN but the problem persists. I had a theory about this problem which I did not try out.

The app [AnkiDroid](https://play.google.com/store/apps/details?id=com.ichi2.anki) requires a folder named 'fonts' in the local storage if one wants to use custom apps for decks. I think this might have been interfering with the library but I'm not sure.

I hope this helps.
